### PR TITLE
feat: adds useActiveProfileSwitcher hook

### PR DESCRIPTION
**Pull request by cesarenaldi**
Also removes unused method.

### CODE REVIEW

File: .changeset/afraid-insects-move.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@lens-protocol/api-bindings": minor
+"@lens-protocol/domain": minor
+"@lens-protocol/react": minor
```
Comment (desfero) [inline .changeset/afraid-insects-move.md:4]: nit: would be better to just mention the package that this change affects the most. Right now changeset will generate the changelog mentioning `Added useProfileOwnedBy hook` in `api-bindings, domain and react` but in reality `useProfileOwnedBy` is only a part of `react` package, and the changes in `api-bindings,domain` are internal one 
(same applies to the `happy-socks-design` changeset)

File: examples/web-wagmi/src/profiles/ProfilesPage.tsx [hunk 2]
```diff
@@ -46,6 +46,16 @@ const profileHooks = [
...
     path: '/profiles/useFollow',
   },
+  {
+    label: 'useProfilesOwnedBy',
+    description: `Fetch a list of profiles owed by a given address.`,
+    path: '/profiles/useProfilesOwnedBy',
+  },
+  {
+    label: 'useActiveProfileSwitch',
+    description: `Switch active profile`,
```
Comment (desfero) [inline examples/web-wagmi/src/profiles/ProfilesPage.tsx:56]:
```suggestion
    description: `Switch active profile.`,
```

File: examples/web-wagmi/src/profiles/UseProfileOwnedBy.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { WhenLoggedInWithProfile, WhenLoggedOut } from '../components/auth/auth';
+import { Loading } from '../components/loading/Loading';
+import { useInfiniteScroll } from '../hooks/useInfiniteScroll';
+import { ProfileCard } from './components/ProfileCard';
+
+type UseProfilesOwnedByProps = {
+  address: EthereumAddress;
+};
+
+function UseProfilesOwnedByActiveWallet({ address }: UseProfilesOwnedByProps) {
```
Comment (desfero) [inline examples/web-wagmi/src/profiles/UseProfileOwnedBy.tsx:13]: nit: naming wise we don't know in this component if `address` is an active wallet

File: packages/domain/src/use-cases/profile/ActiveProfileLoader.ts [hunk 4]
```diff
@@ -1,20 +1,11 @@
...
-  getProfileByHandle(handle: string): Promise<Profile | null>;
-}
-
-export interface IActiveProfileGateway {
-  setActiveProfile(profile: Profile): Promise<void>;
-  getActiveProfile(): Promise<Profile | null>;
-}
-
-export class ActiveProfile {
+export class ActiveProfileLoader {
```
Comment (desfero) [inline packages/domain/src/use-cases/profile/ActiveProfileLoader.ts:8]: good naming

File: packages/react/src/profile/adapters/ProfileGateway.ts [hunk 5]
```diff
@@ -44,4 +44,21 @@ export class ProfileGateway implements IProfileGateway {
...
     });
   }
+
+  async getProfileById(profileId: string): Promise<Profile | null> {
+    const { data } = await this.apolloClient.query<GetProfileQuery, GetProfileQueryVariables>({
+      query: GetProfileDocument,
+      variables: { request: { profileId } },
+    });
+
+    invariant(data, `Could not query profiles by id: ${profileId}`);
```
Comment (desfero) [inline packages/react/src/profile/adapters/ProfileGateway.ts:54]: nit: this is kinda tricky as we are shallowing the original error (the one in the `error` returned from the `apolloClient.query`). Can be confusing later to debug real production errors as only `Could not query profiles by id` message is there without a real reason.
Not saying we have to solve it now, just building awareness.

File: packages/react/src/publication/usePublications.ts [hunk 6]
```diff
@@ -13,7 +13,7 @@ type UsePublicationArgs = PaginatedArgs<{
 export function usePublications({
   profileId,
   observerId,
-  limit,
+  limit = DEFAULT_PAGINATED_QUERY_LIMIT,
```
Comment (desfero) [inline packages/react/src/publication/usePublications.ts:16]: good call

**Review by desfero — CHANGES_REQUESTED** [review]
LGTM, just small nitpicks

File: .changeset/afraid-insects-move.md (hunk 1 above)
Comment (cesarenaldi) [reply .changeset/afraid-insects-move.md:4]: got it

File: examples/web-wagmi/src/profiles/UseProfileOwnedBy.tsx (hunk 3 above)
Comment (cesarenaldi) [reply examples/web-wagmi/src/profiles/UseProfileOwnedBy.tsx:13]: I just assumed cause it's private to this module and only active wallet address is fed into it.

File: packages/react/src/profile/adapters/ProfileGateway.ts (hunk 5 above)
Comment (cesarenaldi) [reply packages/react/src/profile/adapters/ProfileGateway.ts:54]: I agree, and this is not the only place where this happens. I created a ticket to discuss this issue: T-4022

**Review by desfero — APPROVED** [review]

**Pull request merged**